pretyping/evarconv.ml

```diff
@@ -1147,6 +1147,36 @@ and evar_eqappr_x ?(rhs_is_already_stuck = false) flags env evd pbty
           |Some (sk1',sk2'), Success i' -> evar_conv_x flags env i' CONV (Stack.zip i' (term1,sk1')) (Stack.zip i' (term2,sk2'))
           end
 
+        | Proj (p1,c1), Proj(p2,c2) when QProjection.Repr.equal env (Projection.repr p1) (Projection.repr p2) ->
+          begin match ise_stack2 true env evd (evar_conv_x flags) sk1 sk2 with
+          |_, (UnifFailure _ as x) -> x
+          |None, Success i' -> evar_conv_x flags env i' CONV c1 c2
+          |Some _, Success _ -> UnifFailure (evd,NotSameHead)
+          end
+
+        (* Catch the p.c ~= p c' cases *)
```

Comment:
Do we really want to add more code to support the (hopefully eventually removed) compatibility layer for primitive projections?

Reply:
I'm not convinced the problem is solvable anyways, so please proceed.

Reply:
```suggestion
        (* Catch the c.(p) ~= p c' cases *)
```
seems more sensible (even though the 2 syntaxes are actually parsed to the same thing), `p.c` is not related to projections.